### ISSUE TITLE
Any `./koboldcpp.py` command hangs desktop GUI on Arch Linux

### ISSUE THREAD

**Issue by cryptoquick** [LostRuins/koboldcpp#203]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now. (I'm running 28f1196f65eb27a8f231fb83b38c91fb42a1a11a, which is not the very latest commit, but the only commit after that fixes a Mac build error, and that's not related to my issue)
- [x] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

After compiling on Arch Linux, I ran `./koboldcpp.py`, first when providing a model, but later I just ran it without anything for debugging purposes. I expected something to happen but instead...

# Current Behavior

The desktop UI froze, even the clock stopped. The cursor went X-shaped and I could move my mouse and draw boxes over my UI, but everything else was essentially frozen. When I draw boxes, the error quacking noise is made. I would try ctrl-C, but that only worked after I clicked around, and not before. If I ran just the command and then an immediate ctrl-C, that did not close the program; I had to click around, then press ctrl-C to regain access to my desktop. No matter what I do, I can't seem to resolve this.

# Environment and Context

Please provide detailed information about your computer setup. This is important in case the issue is not reproducible except for under certain specific conditions.

* Physical (or virtual) hardware you are using, e.g. for Linux:

```
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         48 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  32
  On-line CPU(s) list:   0-31
Vendor ID:               AuthenticAMD
  Model name:            AMD Ryzen 9 5950X 16-Core Processor
    CPU family:          25
    Model:               33
    Thread(s) per core:  2
    Core(s) per socket:  16
    Socket(s):           1
    Stepping:            0
    Frequency boost:     enabled
    CPU(s) scaling MHz:  74%
    CPU max MHz:         5083.3979
    CPU min MHz:         2200.0000
    BogoMIPS:            6799.62
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpu
                         id extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dn
                         owprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms i
                         nvpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lb
                         rv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succo
                         r smca fsrm
Virtualization features:
  Virtualization:        AMD-V
Caches (sum of all):
  L1d:                   512 KiB (16 instances)
  L1i:                   512 KiB (16 instances)
  L2:                    8 MiB (16 instances)
  L3:                    64 MiB (2 instances)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0-31
Vulnerabilities:
  Itlb multihit:         Not affected
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Not affected
  Retbleed:              Not affected
  Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```

* Operating System, e.g. for Linux:

`Linux hephaestus 6.3.4-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Wed, 24 May 2023 17:43:43 +0000 x86_64 GNU/Linux`

* SDK version, e.g. for Linux:

```
$ python3 --version > Python 3.11.3
$ make --version > GNU Make 4.4.1
$ g++ --version > g++ (GCC) 13.1.1 20230429
```

Also, I'm using GNOME 44.1 with GDM set to X11 for windowing.

# Failure Information (for bugs)

There is no feedback other than after pressing ctrl-C and successfully closing the program, this is printed to the terminal:

```
./koboldcpp.py
^C./koboldcpp.py: line 10: stop_token_max: command not found
./koboldcpp.py: line 12: syntax error near unexpected token `('
./koboldcpp.py: line 12: `class load_model_inputs(ctypes.Structure):'
```

# Steps to Reproduce

After checking out the repo and installing the dependencies for Arch, I had to use the `cmake .` fix in another thread, then ran `make LLAMA_OPENBLAS=1 LLAMA_CLBLAST=1`. I then set `chmod +x ./koboldcpp.py`, and then executed the commands specified above, and ran into the error described above. This is consistently reproducible, nothing I try fixes this, and I'm out of ideas.

# Failure Logs

Please include any relevant log snippets or files. If it works under one configuration but not under another, please provide logs for both configurations and their corresponding outputs so it is easy to see where behavior changes.

Also, please try to **avoid using screenshots** if at all possible. Instead, copy/paste the console output and use [Github's markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) to cleanly format your logs for easy readability.

Example environment info:
```
llama.cpp$ git log | head -1
commit 28f1196f65eb27a8f231fb83b38c91fb42a1a11a

llama.cpp$ lscpu | egrep "AMD|Flags"
egrep: warning: egrep is obsolescent; using grep -E
Vendor ID:                       AuthenticAMD
Model name:                      AMD Ryzen 9 5950X 16-Core Processor
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization:                  AMD-V

llama.cpp$ python3 --version
Python 3.11.3

llama.cpp$ pip list | egrep "torch|numpy|sentencepiece"
numpy                         1.24.3
numpydoc                      1.5.0
torch                         2.0.1

llama.cpp$ make --version | head -1
GNU Make 4.4.1
```

**Comment by LostRuins** [comment]
What ggml model did you attempt to load? *Was it a ggml model?* It sounds like something is going very wrong with the load process, and the program is looping infinitely during the load, perhaps you attempted to load a corrupted model? Or could it be that you do not have sufficient RAM for this model and it has started thrashing as it tries to use swap memory?

Let's try this to debug:
1. Start from a clean slate, check out the latest commit on the repo into a new directory.
2. Start with a small model for testing purposes. This is a very small GPT2 model you can try: https://huggingface.co/concedo/cerebras-111M-ggml/blob/main/cerebras-111m-q4_0-newer.bin it's only 111m parameters, fits in less than 512mb, and should be a good test.
3. Do a simple `make`, do not build with `LLAMA_CLBLAST` or `LLAMA_OPENBLAS`, let's try to get the basics working first.

Once the above is done, try running the script `python koboldcpp.py --nommap --noblas` again with the above model, and describe your console output.

**Comment by cryptoquick** [comment]
Okay, compiled with `commit ea336bfa332301c9f520e08fa7a538cc60dc0c96` and using simple `make`

Running:
`./koboldcpp.py models/cerebras-111m-q4_0-newer.bin`

Produced the same behavior as described above, plus this output, once canceled:

```
^C^C./koboldcpp.py: line 10: stop_token_max: command not found
./koboldcpp.py: line 12: syntax error near unexpected token `('
./koboldcpp.py: line 12: `class load_model_inputs(ctypes.Structure):'
```

But then I realized, the sample command you provided used python to execute, not the self-executing file I assumed it to be.

Running this command that you provided, `python koboldcpp.py --nommap --noblas`, works fine!

**Comment by cryptoquick** [comment]
Thank you for the fast response, closing!

**Comment by LostRuins** [comment]
Haha no problem. Thought the .py file extension might've been a clue lol. Hope the normal models works fine then, you should use remove `--nommap` and `--noblas` for better performance once its working
